### DISCUSSION TITLE
Reactive CRUD Performance: A Case Study

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#29505] [Quarkus Blog/Website]
# Reactive CRUD Performance: A Case Study

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/reactive-crud-performance-case-study/

**Comment by fercomunello** [comment]
Great article! =D

**Comment by GreySpike** [comment]
Good article making the most fundamental question of tuning: what is the context for the results?
My rule of thumb as the manager of a performance team is that if the team can't a) repeat the results consistently (minimum 3) and b) show that CPU or ram or IOPS are maxed out, then the bottleneck has not been found and removed.
Of course, once the app is maxing out CPU/ram/IOPS then I will ask how the load can be reduced.
But discussing and exploring why results are inconsistent often leads to the 'ah-ha' moment that leads to discovering the bottleneck.

**Comment by lferna** [comment]
Good article, being important to validate our results.
In our case, even without reactive aproximation, we have a amazing performance with Quarkus and native images, ~4M requests during Black Friday. Probably, according your final tips we may increase our performance.
Good job.

**Comment by derekm** [comment]
For those wondering, in the end, Quarkus was about *14 times faster* than `{INSERT_FRAMEWORK_HERE}`!!!
